File: services/identity-service/src/main/java/br/com/deltaglobalbank/identity/domain/shared/PasswordHasher.java
package br.com.deltaglobalbank.identity.domain.shared;

import br.com.deltaglobalbank.identity.domain.user.HashedPassword;
import br.com.deltaglobalbank.identity.domain.user.Password;

public interface PasswordHasher {
    HashedPassword hash(Password password);

    boolean matches(String rawPassword, HashedPassword hashedPassword);
}

File: services/identity-service/src/main/kotlin/br/com/deltaglobalbank/identity/domain/shared/PasswordHasher.kt
package br.com.deltaglobalbank.identity.domain.shared

import br.com.deltaglobalbank.identity.domain.user.HashedPassword
import br.com.deltaglobalbank.identity.domain.user.Password

interface PasswordHasher {
    fun hash(password: Password): HashedPassword
    fun matches(rawPassword: String, hashedPassword: HashedPassword): Boolean
}
